tests/DucksNet.IntegrationTests/UsersControllerTests.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using System;
+using System.Net.Http.Json;
+using DucksNet.API.Controllers;
+using DucksNet.API.DTO;
+using DucksNet.Domain.Model;
+using System.Net;
+
+namespace DucksNet.IntegrationTests;
+public class UsersControllerTests : BaseIntegrationTests<UsersController>
+{
+    private const string UserUrl = "api/v1/users";
+
+    [Fact]
+    public async Task When_CreatedUser_Then_ShouldReturnPetInTheGetRequest()
+    {
+        ClearDatabase();
+        //Arrange
+        var sut = CreateSut();
+        //Act 
+        var userResponse = await TestingClient.PostAsJsonAsync(UserUrl, sut);
+        userResponse.EnsureSuccessStatusCode();
+        var getUserResult = await TestingClient.GetAsync(UserUrl);
+        //Assert
+
+        var users = await getUserResult.Content.ReadFromJsonAsync<List<User>>();
+        users.Should().NotBeNull();
+        users!.Count.Should().Be(1);
+        foreach (var user in users!)
+        {
+            user.FirstName.Should().Be(sut.FirstName);
+            user.LastName.Should().Be(sut.LastName);
+            user.Address.Should().Be(sut.Address);
+            user.PhoneNumber.Should().Be(sut.PhoneNumber);
+            user.Email.Should().Be(sut.Email);
+            user.Password.Should().Be(sut.Password);
+        }
+    }
+
+    [Fact]
+    public async Task When_CreateUserWithEmptyFirstName_Then_ShouldFail()
+    {
+        ClearDatabase();
+        //Arrange
+        var sut = CreateInvalidSut(" ", "Cats", "Paris, Bonfamille's Residence", "0700000000", "les_aristocats@yahoo.com", "CatsAreTheBest");
+
+        //Act 
+        var userResponse = await TestingClient.PostAsJsonAsync(UserUrl, sut);
+        //Assert
+        userResponse.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+        var errors = await userResponse.Content.ReadFromJsonAsync<List<string>>();
+        errors.Should().NotBeEmpty();
+        errors.Should().Contain("First name is required.");
+    }
+
+    [Fact]
+    public async Task When_CreateUserWithEmptyLastName_Then_ShouldFail()
+    {
+        ClearDatabase();
+        //Arrange
+        var sut = CreateInvalidSut("Aristo", " ", "Paris, Bonfamille's Residence", "0700000000", "les_aristocats@yahoo.com", "CatsAreTheBest");
```

Comment:
It's kinda funny that you created a helper function for making invalid objects but you still pass in all of the parameters for the constructor :)

Reply:
Pai ca sa ocupe mai putine linii =))